### PR TITLE
Prefer transports with most movement left

### PR DESCRIPTION
**Pull request by ron-murhammer**
Addresses point 4 in #553

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/666)

### CODE REVIEW

File: src/games/strategy/triplea/util/TransportUtils.java [hunk 1]
```diff
@@ -158,13 +151,18 @@ public static int getTransportCost(final Collection<Unit> units) {
...
-  private static List<Unit> sortByTransportCapacityAscending(final Collection<Unit> transports) {
+  private static List<Unit> sortByTransportCapacityAscendingThenMovesDescending(final Collection<Unit> transports) {
     final Comparator<Unit> transportCapacityComparator = new Comparator<Unit>() {
       @Override
       public int compare(final Unit o1, final Unit o2) {
         final int capacityLeft1 = TransportTracker.getAvailableCapacity(o1);
         final int capacityLeft2 = TransportTracker.getAvailableCapacity(o2);
-        return capacityLeft1 - capacityLeft2;
+        if (capacityLeft1 != capacityLeft2) {
+          return Integer.compare(capacityLeft1, capacityLeft2);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/util/TransportUtils.java]: Is this swapping o1 and o2?
When:  capacityLeft2 > capacityLeft1,
we'll get a negative number.

Unless I'm mistaken, which I might be, but `Integer.compare(capacityLeft1, capacityLeft2);` would return +1 when capacityLeft2 is larger?

Also of note, the `1` and `2` are then later swapped for the movement:
`return Integer.compare(movementLeft2, movementLeft1);`

If the ordering is correct, a comment in here to help the reader out to know why we swapped the ordering would be of benefit.

File: src/games/strategy/triplea/util/TransportUtils.java [hunk 2]
```diff
@@ -178,7 +176,7 @@ public int compare(final Unit o1, final Unit o2) {
       public int compare(final Unit o1, final Unit o2) {
         final int cost1 = UnitAttachment.get((o1).getUnitType()).getTransportCost();
         final int cost2 = UnitAttachment.get((o2).getUnitType()).getTransportCost();
-        return cost2 - cost1;
+        return Integer.compare(cost2, cost1);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/util/TransportUtils.java]: Again looks like ordering might be swapped. When cost2 is larger than cost1, before we had a positive number. Now with the compare, we'll get a negative number.

File: src/games/strategy/triplea/util/TransportUtils.java (hunk 1 above)
Comment (ron-murhammer) [inline src/games/strategy/triplea/util/TransportUtils.java]: @DanVanAtta I think you are mistaken as the ordering based on capacity isn't change: https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#compare(int,%20int)

File: src/games/strategy/triplea/util/TransportUtils.java (hunk 2 above)
Comment (ron-murhammer) [inline src/games/strategy/triplea/util/TransportUtils.java]: Same as above.

Comment (DanVanAtta) [inline src/games/strategy/triplea/util/TransportUtils.java]: drawing out the truth table,  indeed you were correct the first time.

**Pull request merged**